ReactAndroid/src/main/java/com/facebook/react/uimanager/JSTouchDispatcher.java

```diff
@@ -30,6 +30,7 @@
   private int mTargetTag = -1;
   private final float[] mTargetCoordinates = new float[2];
   private boolean mChildIsHandlingNativeGesture = false;
+  private long mGestureStartTime;
```

Comment:
Is having a single field for this enough? Can it happen that there are two gestures (with two different fingers) happening at once?

finger1 DOWN
finger1 MOVE
finger2 DOWN -- overwrites `mGestureStartTime`
finger2 MOVE
finger1 MOVE -- `mGestureStartTime` is the start time of gesture for finger2, so using `mGestureStartTime` here is wrong